scrapy/utils/misc.py

```diff
@@ -73,6 +101,23 @@ def walk_modules(path):
     return mods
 
 
+def get_module_from_filepath(path):
+    """Load and return a python module/package from a file path"""
+    path = path.rstrip("/")
+    if path.endswith('.py'):
+        path = path.rsplit('.py', 1)[0]
+    basefolder, modname = os.path.split(path)
+    # XXX: There are other ways to import modules from a full path which don't
+    #      need to modify PYTHONPATH, see
+    #          https://stackoverflow.com/questions/67631/
+    #      These methods differ between py2 and py3, and apparently the
+    #      py3 method was deprecated in Python 3.4
+    sys.path.insert(0, basefolder)
+    mod = import_module(modname)
+    sys.path.pop(0)
```

Comment:
Indeed! This could be annoying to maintain, I now think it's best to drop support for giving a file path as add-on. Users should just load the module themselves if they absolutely don't want to (or cannot) use Python paths.